seed/seedwriter/manifest.go

```diff
@@ -93,27 +93,27 @@ func (vs *SeedManifestValidationSet) hasSnap(snapName string) bool {
 // <account-id>/<name>=<sequence>
 // <account-id>/<name> <sequence>
 // <snap-name> <snap-revision>
-type SeedManifest struct {
-	revsAllowed map[string]*SeedManifestSnapRevision
-	revsSeeded  map[string]*SeedManifestSnapRevision
-	vsAllowed   map[string]*SeedManifestValidationSet
-	vsSeeded    map[string]*SeedManifestValidationSet
+type Manifest struct {
+	revsAllowed map[string]*ManifestSnapRevision
+	revsSeeded  map[string]*ManifestSnapRevision
+	vsAllowed   map[string]*ManifestValidationSet
+	vsSeeded    map[string]*ManifestValidationSet
 }
 
-func NewSeedManifest() *SeedManifest {
-	return &SeedManifest{
-		revsAllowed: make(map[string]*SeedManifestSnapRevision),
-		revsSeeded:  make(map[string]*SeedManifestSnapRevision),
-		vsAllowed:   make(map[string]*SeedManifestValidationSet),
-		vsSeeded:    make(map[string]*SeedManifestValidationSet),
+func NewManifest() *Manifest {
+	return &Manifest{
+		revsAllowed: make(map[string]*ManifestSnapRevision),
+		revsSeeded:  make(map[string]*ManifestSnapRevision),
+		vsAllowed:   make(map[string]*ManifestValidationSet),
+		vsSeeded:    make(map[string]*ManifestValidationSet),
 	}
 }
 
-// NewSeedManifestForTest is stricly for unit tests, do not use for non-test code.
-func NewSeedManifestForTest(revsAllowed, revsSeeded map[string]*SeedManifestSnapRevision, vsAllowed, vsSeeded map[string]*SeedManifestValidationSet) *SeedManifest {
-	osutil.MustBeTestBinary("NewSeedManifestForTest can only be used in unit tests")
+// NewManifestForTest is stricly for unit tests, do not use for non-test code.
+func NewManifestForTest(revsAllowed, revsSeeded map[string]*ManifestSnapRevision, vsAllowed, vsSeeded map[string]*ManifestValidationSet) *Manifest {
```

Comment:
unrelated but MockManifest would have been a more typical name for this btw

Reply:
Ah ok! I'll rename this, thanks for the review!